apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/logging/core/FileWriter.java

```diff
@@ -212,15 +212,11 @@ private boolean prepareWriteStream() {
     }
 
     /**
-     * Write log to the queue. W/ performance trade off, set 2ms timeout for the log OP.
+     * Write log to the queue. W/ performance trade off.
      *
      * @param message to log
      */
     @Override public void write(String message) {
-        try {
-            logBuffer.offer(message, 2, TimeUnit.MILLISECONDS);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        }
+        logBuffer.offer(message);
```

Comment:
After investing the JDK source codes, this will cause more lock time than original. Because this will never release lock until `offer` successfully.

Reply:
`queue.offer(element)` will insert when the queue has free space. If not free space, the element will be discarded and release the lock immediately.
As you description, I think you say the operator is `queue.put(element)`.

Reply:
I just rechecked, it seems yes. I misread the codes last time. Then this PR should be good enough.

But as 6.6.0 codes have been frozen, this will only be merged after 6.6.0 release.